GH-13: Added ability to show checked files (resolves #13).

the-t-in-rtf opened the pull request:
See #13 for details.  Note that as part of this I refactored so that all checks use a common gateway function to:

1. Confirm whether they should run.
2. Lookup their files in a standard fashion, including making a list of checked files.

amb26 commented on src/js/jsonEslint.js:
```diff
@@ -30,81 +30,72 @@ fluid.defaults("fluid.lintAll.jsonEslint", {
+            fluid.each(eslintResults, function (singleError) {
+                var relativePath = path.relative(that.options.rootPath, singleError.filePath);
+
+                var fileErrorCount    = fluid.get(singleError, "errorCount");
```
odd spacing here and blank line on line 76

amb26 commented on src/js/jsonEslint.js:
```diff
@@ -30,81 +30,72 @@ fluid.defaults("fluid.lintAll.jsonEslint", {
+            });
+            wrappedPromise.resolve(that.results);
+        },
+        wrappedPromise.reject
```
This could probably be https://docs.fluidproject.org/infusion/development/promisesapi#fluidpromisemapsource-func ?

the-t-in-rtf replied:
Agreed.  I reviewed other uses of wrapped promises, and left them alone, as they're either wrapping non-Fluid promises or needed to catch and report low-level errors.

amb26 commented on src/js/jsonEslint.js:
```diff
@@ -30,81 +30,72 @@ fluid.defaults("fluid.lintAll.jsonEslint", {
+                fluid.each(singleError.messages, function (singleMessage) {
+                    // Our "wrapper" adds three lines of code before the first line.
+                    var adjustedLine = singleMessage.line - 3;
+                    // Our "wrapper" adds 17 characters to the first line.
```
What are the 17 characters?

the-t-in-rtf replied:
I'll update the comment to refer people to the named variable.

amb26 commented on src/js/json5lint.js:
```diff
@@ -24,40 +24,32 @@ fluid.defaults("fluid.lintAll.json5lint", {
-    }
+fluid.lintAll.json5lint.runChecks = function (that, filesToScan) {
+
+    // TODO: Consider using parse instead if it provides better feedback on where failures are.
```
Comment is out of date?

the-t-in-rtf replied:
Yes, that's left over from before I made the exact change referenced.

amb26 commented on src/js/json5lint.js:
```diff
@@ -24,40 +24,32 @@ fluid.defaults("fluid.lintAll.json5lint", {
  * @param {Object} that - The `fluid.lintAll.json5lint` component.
- * @param {Array<String>} [checksToRun] - An array of check "keys" indicating which checks should be run.  If omitted,
- * all checks are run.
+ * @param {Array<String>} filesToScan - An array of files to check.
```
I can't remember if our preferred form for these was String[] ?

the-t-in-rtf replied:
I think we've used a mix, for example here's one with the square notation in [core Infusion](https://github.com/fluid-project/infusion/blob/main/src/framework/core/js/Fluid.js#L460).  However, we also used `Array` on its own in the same file, instead of using square brackets.  I don't have a strong opinion about which syntax to use, I think we should:

1. Discuss at the next infusion-related meeting or on the chat/mailing list.
2. Agree on a convention.
3. Make it possible to check our work using [additional configuration options for the jsdoc plugin for ESLint](https://github.com/gajus/eslint-plugin-jsdoc/blob/master/test/rules/assertions/checkTypes.js#L661).

the-t-in-rtf commented:
@amb26, I have addressed the feedback you left.  I think we should hold off on the `Array<String>` change until we can agree on a standard and make a separate ticket to push through an update to `eslint-config-fluid`.

amb26 commented:
Merged - branch coverage is a little low because the launcher is untested

the-t-in-rtf commented:
> Merged - branch coverage is a little low because the launcher is untested

I filed #16 for that.
